src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -781,7 +781,13 @@ void CompilerOracle::parse_from_line(char* line) {
           print_parse_error(error_buf, original.get());
           return;
         }
-        register_command(typed_matcher, option, true);
+        if (option2type(option) == OptionType::Bool) {
+          register_command(typed_matcher, option, true);
+        } else {
+          jio_snprintf(error_buf, sizeof(error_buf), "  Option '%s' is not a boolean type", option2name(option));
```

Comment:
I don't think it is correct message for such option. The example specifies value after flag's name - in such case it is missing type:
`option,Test::test,double,TestOptionDouble,3.14`

Is it possible to produce more accurate error message about missing specified type?

Reply:
> I don't think it is correct message for such option. The example specifies value after flag's name - in such case it is missing type:
> `option,Test::test,double,TestOptionDouble,3.14`
> 
> Is it possible to produce more accurate error message about missing specified type?

Thanks @vnkozlov for your review.

The updated err_msg is

```
Missing type 'double' before option 'TestOptionDouble'
```

Reply:
> I don't think it is correct message for such option. The example specifies value after flag's name - in such case it is missing type:
> `option,Test::test,double,TestOptionDouble,3.14`
> 
> Is it possible to produce more accurate error message about missing specified type?

It's correct since bool is assumed when the type is left out. But the newly suggested message is even better.